### PR TITLE
add skeleton docs/scripts for 1.3 upgrade

### PR DESCRIPTION
**Pull request by dzou-hpe**
## Summary and Scope

Skeleton docs/script for 1.3 upgrade

# NOTE:
this is only a skeleton and any broken links/wrong path/lint is out of scope. In next few PRs, we will need to 
1. pull all fixes from 1.2 to 1.3 where it makes sense
2. run 1.3 upgrade and fix whatever is broken
3. remove steps that are not required in 1.3

Bottom line: this is just a skeleton so we can actually work on 1.3 upgrade. 

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves CASMINST-4655

## Testing

_List the environments in which these changes were tested._

### Tested on:

No test, as it is a skeleton

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: upgrade/management_nodes_and_csm_services/scripts/upgrade/prepare-assets.sh [hunk 1]
```diff
@@ -81,7 +81,7 @@ if [[ -z ${TARBALL_FILE} ]]; then
 
     if [[ -z ${ENDPOINT} ]]; then
         # default endpoint to internal artifactory
-        ENDPOINT=https://artifactory.algol60.net/artifactory/releases/csm/1.2/
+        ENDPOINT=https://artifactory.algol60.net/artifactory/releases/csm/management_nodes_and_csm_services/
```
Comment (mharding-hpe) [inline upgrade/management_nodes_and_csm_services/scripts/upgrade/prepare-assets.sh:84]: Are we changing the URLs on artifactory?

File: upgrade/management_nodes_and_csm_services/Stage_0_Prerequisites.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ backup of Workload Manager configuration data and files is created. Once complet
    > **Important:** The upgrade scripts expect the `docs-csm` RPM to be located at `/root/docs-csm-latest.noarch.rpm`; that is why this command copies it there.
 
    ```bash
-   ncn-m001# wget https://artifactory.algol60.net/artifactory/csm-rpms/hpe/stable/sle-15sp2/docs-csm/1.2/noarch/docs-csm-latest.noarch.rpm \
+   ncn-m001# wget https://artifactory.algol60.net/artifactory/csm-rpms/hpe/stable/sle-15sp2/docs-csm/management_nodes_and_csm_services/noarch/docs-csm-latest.noarch.rpm \
```
[the remark this answers is not in the document]
Comment (mharding-hpe) [reply upgrade/management_nodes_and_csm_services/Stage_0_Prerequisites.md:45]: This was a case where you actively made an edit -- I am just wondering if this was a deliberate edit. I didn't think we'd be changing the URL to the docs RPM on artifactory.

[the remark this answers is not in the document]
Comment (mharding-hpe) [reply upgrade/management_nodes_and_csm_services/Stage_0_Prerequisites.md:45]: Unless this IS the change you intended to make, it seems weird to replace a working command with one we know is broken.

I think you should absolutely remove any content that is not going to be used for csm-1.3 forward.

If there is content that will still be used but just needs to be modified to reflect the repo directory structure, then I think that should stay.

But again, this wget command doesn't seem related to the docs-csm internal repo directory structure, unless I'm missing something.

[the remark this answers is not in the document]
Comment (mharding-hpe) [reply upgrade/management_nodes_and_csm_services/Stage_0_Prerequisites.md:45]: If you just want a place to begin testing out changes, then I think a development branch is the right place for that. You can always copy that onto a system to test upgrades -- it doesn't need to go through the regular build process while you're still working out the initial kinks.

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
This seems like it should be a development branch, not merged into main. In theory we shouldn't be merging incomplete or broken stuff into a release branch or into main. Is there a reason we need this to be in main rather than in a development branch? We still build RPMs from the development branch and put them out on artifactory, so we'd still be able to test even using the dev branch, I'd think.

**Comment by dzou-hpe** [comment]
> This seems like it should be a development branch, not merged into main. In theory we shouldn't be merging incomplete or broken stuff into a release branch or into main. Is there a reason we need this to be in main rather than in a development branch? We still build RPMs from the development branch and put them out on artifactory, so we'd still be able to test even using the dev branch, I'd think.

i like the development branch idea :) but... anyways, i am waiting for news on should I even work on this PR given that Jim is trying to push through a 1.3 upgrade with existing doc. i will update the PR.

File: upgrade/management_nodes_and_csm_services/Stage_0_Prerequisites.md [hunk 3]
```diff
@@ -96,7 +96,7 @@ backup of Workload Manager configuration data and files is created. Once complet
 1. Run the script.
 
    ```bash
-   ncn-m001# /usr/share/doc/csm/upgrade/1.2/scripts/upgrade/prepare-assets.sh --csm-version ${CSM_RELEASE} --tarball-file "${CSM_TAR_PATH}"
+   ncn-m001# /usr/share/doc/csm/upgrade/management_nodes_and_csm_services/scripts/upgrade/prepare-assets.sh --csm-version ${CSM_RELEASE} --tarball-file "${CSM_TAR_PATH}"
```
Comment (rustydb) [inline upgrade/management_nodes_and_csm_services/Stage_0_Prerequisites.md:99]: Can you please fix-up these prompts to match the `main` branch convention? https://github.com/Cray-HPE/docs-csm/blob/MTL-1695/introduction/documentation_conventions.md#step-examples

**Review by rustydb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**